sch_yang_condition.c

```diff
@@ -0,0 +1,923 @@
+                                     depth, flags);
+            presult->result = result.result;
+            if (result.result)
+            {
+                presult->value = result.value;
+                result.value = NULL;
+            }
+        }
+        else if (g_strcmp0 (xnode->name, "derived-from") == 0)
+        {
+            sch_yang_function_derived_from (instance, root, path, step_path, xnode,
+                                            &result, depth, false, flags);
+            presult->result = result.result;
+        }
+        else if (g_strcmp0 (xnode->name, "derived-from-or-self") == 0)
+        {
+            sch_yang_function_derived_from (instance, root, path, step_path, xnode,
+                                            &result, depth, true, flags);
+            presult->result = result.result;
+        }
+        else if (g_strcmp0 (xnode->name, "current") == 0)
+        {
+            presult->value = g_strdup (path);
+            presult->result = true;
+        }
+        else if (g_strcmp0 (xnode->name, "name") == 0)
+        {
+            sch_yang_function_name (instance, root, path, step_path, xnode, &result,
+                                    depth, flags);
+            *presult = result;
+            result.value = NULL;
+        }
+        else if (g_strcmp0 (xnode->name, "if-feature") == 0)
+        {
+            sch_yang_function_if_feature (instance, root, path, step_path, xnode,
+                                          &result, depth, flags);
+            *presult = result;
+            result.value = NULL;
+        }
+        else if (g_strcmp0 (xnode->name, "boolean") == 0)
+        {
+            sch_yang_process_arg_list (instance, root, path, step_path, xnode, &result,
+                                       depth, flags);
+            presult->result = result.result;
+        }
+        break;
+    case XPATH_TYPE_STEP:
+        if (xnode->left)
+        {
+            new_step_path = g_strdup (step_path ? : path);
+            sch_yang_process_xnode (instance, root, path, new_step_path, xnode->left,
+                                    &result, depth + 1, flags);
+            presult->result = result.result;
+            if (result.result)
+            {
+                presult->value = result.value;
+                result.value = NULL;
+                if (presult->value)
+                {
+                    g_free (new_step_path);
+                    new_step_path = g_strdup (presult->value);
+                }
+            }
+            else
+            {
+                g_free (result.value);
+                result.value = NULL;
+            }
+
+            if (xnode->right)
+            {
+                sch_yang_process_xnode (instance, root, path, new_step_path, xnode->right,
+                                        &result, depth + 1, flags);
+                presult->result = presult->result & result.result;
+
+                g_free (_path);
```

Comment:
_path is only referenced once above, where it is initialised to NULL - doesn't seem right.

Reply:
Removed _path from the function.